### PR TITLE
Replaced coveralls with codecov

### PR DESCRIPTION
**Pull request by psturc**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-7810
## What
Publishing code coverage results with **coveralls** was recently a bit flaky (was randomly failing) - we've decided to go with **codecov** to see if it improves stability.
## How
* test coverage setup is done by following the steps in https://github.com/istanbuljs/nyc#integrating-with-codecov
* codecov token is stored in environment variable in circleci project
## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Additional Notes
Assuming that if user would start developing by cloning this repo first, the name of the folder he/she would work in would be named 'data-sync-server'. By following the steps in readme,  running `docker-compose up` would create container names prefixed with the folder name.
For this reason, it makes sense to change the prefix in scripts to 'data-sync-server'.
Wdyt?

### CODE REVIEW

**Comment by camilamacedo86** [comment]
@psturc , so codecov is now the standard and should be used in all projects? 

c/c @darahayes ( if yes, we need to update the doc to centralized and define the standards)

File: package.json [hunk 1]
```diff
@@ -35,9 +35,9 @@
     "release:validate": "./scripts/validateRelease.sh",
     "db:init": "FORCE_DROP=true node ./scripts/sync_models",
     "db:init:memeo:inmem": "FORCE_DROP=true node ./scripts/sync_models && sequelize db:seed --seed memeolist-example-inmem.js",
-    "db:init:memeo:postgres": "FORCE_DROP=true node ./scripts/sync_models && sequelize db:seed --seed memeolist-example-postgres.js && docker exec datasyncserver_postgres_memeo_1 psql -U postgres -d memeolist_db -f /tmp/examples/memeolist.tables.sql",
-    "db:shell": "docker exec -it datasyncserver_postgres_1 psql -U postgresql -d aerogear_data_sync_db",
-    "db:shell:memeo": "docker exec -it datasyncserver_postgres_memeo_1 psql -U postgresql -d memeolist_db"
+    "db:init:memeo:postgres": "FORCE_DROP=true node ./scripts/sync_models && sequelize db:seed --seed memeolist-example-postgres.js && docker exec data-sync-server_postgres_memeo_1 psql -U postgres -d memeolist_db -f /tmp/examples/memeolist.tables.sql",
```
Comment (darahayes) [inline package.json]: @psturc can you please revert these particular changes and refer to [this comment](https://github.com/aerogear/data-sync-server/pull/63#discussion_r212236201) on #63

**Comment by darahayes** [comment]
@psturc let's sort out the docker-compose stuff in a separate PR!

**Comment by darahayes** [comment]
@camilamacedo86 I don't know if it makes sense to say 'We must use this one particular service.' It depends on what makes sense for that particular project. Perhaps instead we can have a section with suggested cove coverage services and then links to reference implementations within our own org.

This repo could be a reference implementation for Node.js + Codecov.io
The [aerogear-app-metrics](https://github.com/aerogear/aerogear-app-metrics) could be a reference implementation for Golang + coveralls.

How does that sound? 😃

**Pull request merged**